index.js

```diff
@@ -1,5 +1,15 @@
-import { requireNativeComponent } from 'react-native';
+import React from "react";
+import { requireNativeComponent, View } from "react-native";
 
-const ContextMenu = requireNativeComponent('ContextMenu', null);
+const NativeContextMenu = requireNativeComponent("ContextMenu", null);
+
+const ContextMenu = (props) => {
+  return (
+    <NativeContextMenu {...props}>
+      <View>{props.children}</View>
```

Comment:
What happens when view flattening is turned on for iOS?

I’d be tempted to flip this - don’t wrap the main children, if there’s a preview, wrap that but set the [nativeID](https://reactnative.dev/docs/view#nativeid) prop, which will disable view flattening, and you can [read in obj c](https://github.com/facebook/react-native/blob/7b05b091fd97f95b778369277ac2147730abc7b8/React/Views/UIView%2BReact.m#L38) to check if it’s the preview

Reply:
Hmm you got a link about the view flattening? Haven't read about that yet

Reply:
https://reactnative.dev/architecture/view-flattening

In short, it won't produce UIViews for elements that are only there for layout - which is the case here